specification/botservice/resource-manager/Microsoft.BotService/preview/2017-12-01/botservice.json

```diff
@@ -1618,6 +1982,202 @@
         "siteName"
       ]
     },
+    "ConnectionItemName": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Connection Item name that has been added in the API",
+          "readOnly": true
+        }
+      },
+      "description": "The display name of a connection Item Setting registered with the Bot"
+    },
+    "ConnectionSettingParameter": {
+      "properties": {
+        "key": {
+          "type": "string",
+          "description": "Key for the Connection Setting Parameter.",
+          "readOnly": true
+        },
+        "value": {
+          "type": "string",
+          "description": "Value associated with the Connection Setting Parameter.",
+          "readOnly": true
+        }
+      },
+      "description": "Extra Parameter in a Connection Setting Properties to indicate service provider specific properties"
+    },
+    "ConnectionSettingProperties": {
+      "properties": {
+        "clientId": {
+          "type": "string",
+          "description": "Client Id associated with the Connection Setting."
+        },
+        "clientSecret": {
+          "type": "string",
+          "description": "Client Secret associated with the Connection Setting"
+        },
+        "scopes": {

```

Comment:
what can the scopes be?

Reply:
scopes are serviceprovider dependent and are comma-separated or whitespace separated.
for example, an example scope string for aad is "email.read email.write profile.read" the specific service provider logic knows how to handle this.